core/src/main/java/com/epam/esm/dao/impl/TagDaoImpl.java

```diff
@@ -0,0 +1,108 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.dao.exception.TagNotFoundException;
+import com.epam.esm.model.entity.TagEntity;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Component;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.util.List;
+
+@Component
+public class TagDaoImpl implements TagDao {
+    private final JdbcTemplate jdbcTemplate;
+
+
+    private static final RowMapper<TagEntity> TAG_ROW_MAPPER = (ResultSet resultSet, int rowNum) -> {
+        return new TagEntity(
+                resultSet.getLong("id"),
+                resultSet.getString("name"));
+    };
+
+    private static final String COLUMN_ID = "id";
+
+    private static final int ZERO_INDEX = 0;
+
+    private static final String FIND_ALL_TAGS_QUERY = "select * from tag";
+    private static final String FIND_TAG_BY_ID_QUERY = "select * from tag where id = ?";
+    private static final String DELETE_TAG_BY_ID = "delete from tag where id = ?";
+    private static final String INSERT_TAG_QUERY = "insert into tag (name) values (?)";
+    private static final String SELECT_TAGS_BY_NAME = "SELECT * FROM tag WHERE name = ?";
+    private static final String SELECT_TAGS_BY_PART_NAME = "SELECT * FROM tag WHERE name like ?";
```

Comment:
could you try to use wildcard into sql? I mean smth like "SELECT * FROM tag WHERE name like '%?%'"
I'm not sure that it is possible

Reply:
This approach not working